### PR TITLE
FEATURE: Add periodic problem checks for each LLM in use

### PR DESCRIPTION
**Pull request by keegangeorge**
### 🔍 Overview
This feature adds a periodic problem check which periodically checks for issues with LLMs that are in use. Periodically, we will run a test to see if the in use LLMs are still operational. If it is not, the LLM with the problem is surfaced to the admin so they can easily go and update the configuration.

### 📸 Screenshots
Dashboard Problem Check AI

### CODE REVIEW

File: app/services/problem_check/ai_llm_status.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+class ProblemCheck::AiLlmStatus < ProblemCheck
+  self.priority = "high"
+  self.perform_every = 1.hour
```
[the remark this answers is not in the document]
Comment (romanrizzi) [reply app/services/problem_check/ai_llm_status.rb:5]: I don't think it's very expensive, but maybe it's too much? 3/6 hours might be a better target.

File: config/locales/server.en.yml [hunk 2]
```diff
@@ -453,3 +453,6 @@ en:
       no_default_llm: The persona must have a default_llm defined.
       user_not_allowed: The user is not allowed to participate in the topic.
       prompt_message_length: The message %{idx} is over the 1000 character limit.
+  dashboard:
+    problem:
+      ai_llm_status: "The LLM model: %{model_name} is encountering issues. Please check the <a href='/admin/plugins/discourse-ai/ai-llms/%{model_id}'>model's configuration</a>."
```
Comment (pmusaraj) [inline config/locales/server.en.yml:458]:
```suggestion
      ai_llm_status: "The LLM model: %{model_name} is encountering issues. Please check the <a href='/admin/plugins/discourse-ai/ai-llms/%{model_id}'>model's configuration page</a>."
```

File: app/services/problem_check/ai_llm_status.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    return [] if !SiteSetting.discourse_ai_enabled
+
+    LlmModel.in_use.find_each.filter_map do |model|
+      try_validate(model) { validator.run_test(model) }
+    end
+  end
+
+  def try_validate(model, &blk)
+    begin
+      raise({ message: "Forced error for testing" }.to_json) if Rails.env.test?
```
Comment (pmusaraj) [inline app/services/problem_check/ai_llm_status.rb:23]: Hmm... this means you can't test the use case for "model is enabled and working fine" though.

Comment (pmusaraj) [reply app/services/problem_check/ai_llm_status.rb:23]: Does the dummy model have an auth mechanism? One way I've seen this tested in other places (login methods for example) is by mocking the author response, if it doesn't return a 200, then the problem check is raised.

Comment (pmusaraj) [reply app/services/problem_check/ai_llm_status.rb:23]: In other words, if the credentials are wrong, will the problem check be raised here?

Comment (romanrizzi) [reply app/services/problem_check/ai_llm_status.rb:23]: You will have to config a real model and stub an HTTP request to simulate the error case. Set one of the module's site setting to that model and run the test.

File: config/locales/server.en.yml [hunk 4]
```diff
@@ -453,3 +453,6 @@ en:
       no_default_llm: The persona must have a default_llm defined.
       user_not_allowed: The user is not allowed to participate in the topic.
       prompt_message_length: The message %{idx} is over the 1000 character limit.
+  dashboard:
+    problem:
+      ai_llm_status: "The LLM model: %{model_name} is encountering issues. Please check the <a href='%{base_path}/admin/plugins/discourse-ai/ai-llms/%{model_id}'>model's configuration page</a>."
```
Comment (gschlager) [inline config/locales/server.en.yml:458]: BTW, it's actually recommended to build the URL in Ruby, and have a single %{url} parameter. If the URL ever changes, translators don't need to update their translations.

**Review by romanrizzi — APPROVED** [review]

File: config/locales/server.en.yml (hunk 4 above)
Comment (davidtaylorhq) [reply config/locales/server.en.yml:458]: I wonder if it's the way we cache problem checks? I ran into it earlier in the week when working on encrypt: (internal `/t/128444/7`)

File: app/services/problem_check/ai_llm_status.rb [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  end
+
+  def try_validate(model, &blk)
+    begin
+      blk.call
+      nil
+    rescue => e
+      error_message = parse_error_message(e.message)
+      message =
+        "#{I18n.t("dashboard.problem.ai_llm_status", { base_path: base_path, model_name: model.display_name, model_id: model.id })}"
```
Comment (Drenmi) [inline app/services/problem_check/ai_llm_status.rb:31]: Sorry. Late to the party. This shouldn't really need to be wrapped in an interpolation string? 🙂

**Pull request merged**